### PR TITLE
[Docs] Update links from WooCommerce.com to Woo.com for the Woo Blocks documentation

### LINKED ISSUE #43052: [DOC-BUG] Update links from WooCommerce.com to Woo.com for the Woo Blocks documentation
## Description of the Suggestion

Now that the WooCommerce Blocks plugin was merged to the Woo Monorepo I suggest updating all links within the documentation or source code that currently point to WooCommerce.com to instead point to the new domain, Woo.com. This includes any links within text, code examples, or resources.

 ## Reason for the Suggestion

 WooCommerce has recently rebranded and moved to the new domain, Woo.com. To ensure consistency, I believe it would be great to update all documentation references to reflect this change.

 ## Additional Context

Official announcement of the rebrand: <https://woo.com/posts/say-hello-to-woo-com/>
The files to be updated are inside: https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks

### PR DESCRIPTION
**Pull request by thealexandrelara**
### Changes proposed in this Pull Request:

Now that the WooCommerce Blocks plugin was merged to the Woo Monorepo I suggest updating all links within the documentation or source code that currently point to WooCommerce.com to instead point to the new domain, Woo.com. This includes any links within text, code examples, or resources.

Closes #43052 .

### How to test the changes in this Pull Request:
Review updated files to ensure the links are correctly updated to Woo.com.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 
Update links from WooCommerce.com to Woo.com for the Woo Blocks documentation

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/.github/patch-initial-checklist.md
Comment (Aljullu) [inline plugins/woocommerce-blocks/.github/patch-initial-checklist.md:1]: No need to do it in this PR, but we can probably remove the release checklists now, no? :thinking:

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md [hunk 1]
```diff
@@ -282,4 +282,4 @@ private function save_shipping_instructions() {
 
 ## Conclusion
 
-By following the steps above, you can add and process new field blocks in the WooCommerce checkout block. For complete implementation and additional examples, refer to the provided [tutorial](https://developer.woocommerce.com/2023/08/07/extending-the-woocommerce-checkout-block-to-add-custom-shipping-options/) and the corresponding [GitHub repository](https://github.com).
+By following the steps above, you can add and process new field blocks in the WooCommerce checkout block. For complete implementation and additional examples, refer to the provided [tutorial](https://developer.woo.com/2023/08/07/extending-the-woocommerce-checkout-block-to-add-custom-shipping-options/) and the corresponding [GitHub repository](https://github.com).
```
Comment (Aljullu) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md]: Not related to this PR, but should this link point to the repo mentioned above?
```suggestion
By following the steps above, you can add and process new field blocks in the WooCommerce checkout block. For complete implementation and additional examples, refer to the provided [tutorial](https://developer.woo.com/2023/08/07/extending-the-woocommerce-checkout-block-to-add-custom-shipping-options/) and the corresponding [GitHub repository](https://github.com/woocommerce/wceu23-shipping-workshop-final/).
```

**Review by Aljullu — APPROVED** [review]
LGTM, thanks for taking care of this @thealexandrelara! I left a couple of comments but they are not directly related to this PR, so feel free to disregard them.

File: plugins/woocommerce-blocks/.github/patch-initial-checklist.md
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/.github/patch-initial-checklist.md:1]: I believe we can remove these checklists, I believe we won't be using them anymore

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md (hunk 1 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md]: Nice catch! I agree, just included your suggestion to the PR!

**Pull request merged**